### PR TITLE
[engine] avoid exceptions in speedy control

### PR DESCRIPTION
**Pull request by nickchapman-da**
Simplify speedy control:
- (Fix nitpicks in previous PR)
- prefer using simple `Control` return value to `raise` exception, for normal execution control flow
- add: `Control.Question` and `Control.Complete`
- replacing `SpeedyHungy` and `SpeedyComplete`

In a future PR, I would like to avoid `raise SError...` also, but this will be a bigger change.

### CODE REVIEW

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala [hunk 1]
```diff
@@ -531,38 +528,33 @@ private[lf] object Speedy {
...
+            case Control.Question(res: SResult) =>
+              res
+            case Control.Complete(value: SValue) =>
+              if (enableInstrumentation) track.print()
+              ledgerMode match {
+                case OffLedger => SResultFinal(value, None)
+                case onLedger: OnLedger =>
+                  val ctx = onLedger.ptx.finish
+                  SResultFinal(value, Some(ctx))
+              }
```
Comment (akrmn) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala:552]: nice!

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala [hunk 2]
```diff
@@ -1670,18 +1662,6 @@ private[lf] object Speedy {
     }
   }
 
-  /** Internal exception thrown when a continuation result needs to be returned.
```
Comment (akrmn) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala:1673]: # :fire:

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala [hunk 3]
```diff
@@ -1109,11 +1101,11 @@ private[lf] object Speedy {
 
   private[speedy] sealed abstract class Control
   object Control {
+    final case object WeAreUnset extends Control
     final case class Expression(e: SExpr) extends Control
     final case class Value(v: SValue) extends Control
-    final case class WeAreUnset() extends Control
-    final case class WeAreComplete() extends Control
-    final case class WeAreHungry(res: SResult) extends Control
```
Comment (akrmn) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala:1116]: RIP
`Control.{WeAreComplete,WeAreHungry}`
2022-08-04 - 2022-08-05

**Review by akrmn — APPROVED** [review]
awesome! thank you Nick

**Review by carlpulley-da — APPROVED** [review]
Nice to see these exceptions going. We're at risk here of being able to run Speedy in an async manner!

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala (hunk 3 above)
Comment (nickchapman-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala:1116]: This is very funny! :smile:

**Pull request merged**